sanic/server.py

```diff
@@ -291,17 +294,27 @@ def on_headers_complete(self):
                 self.request
             )
             if self._is_stream_handler:
-                self.request.stream = asyncio.Queue()
+                self.request.stream = StreamBuffer(
+                    self.request_buffer_queue_size
+                )
                 self.execute_request_handler()
 
     def on_body(self, body):
         if self.is_request_stream and self._is_stream_handler:
             self._request_stream_task = self.loop.create_task(
-                self.request.stream.put(body)
+                self.body_append(body)
             )
             return
```

Comment:
Why do we have this `return` here since the caller is not expecting a return? Seems a simpler `if/else` construct would be more self-explanatory.

Reply:
yeah, agree